pkgs/build-support/bintools-wrapper/ld-wrapper.sh

```diff
@@ -57,8 +57,8 @@ fi
 
 source @out@/nix-support/add-hardening.sh
 
-extraAfter=("${hardeningLDFlags[@]}")
-extraBefore=()
+extraAfter=()
```

Comment:
@oxij @FRidh if we do all of this PR *but* this part, we should have the same behavior as before. That seems like a good start to me

Reply:
Sounds good.

Reply:
I really need to go to sleep but somebody else feel free to try it.

Reply:
Can the hardening flags be injected at the start instead of the end? I'd hate to see us reintroduce the problems that prompted this PR in the first place.